### PR TITLE
Fix preemption tests that use PDB

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
Scheduler integration tests that test preemption in presence of PDB had an issue causing PDB status not getting updated. This PR fixes the issue.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig scheduling

xref/ #57057

### CODE REVIEW

File: test/integration/scheduler/preemption_test.go [hunk 1]
```diff
@@ -793,13 +827,31 @@ func TestPDBInPreemption(t *testing.T) {
...
 		// Create PDBs.
 		for _, pdb := range test.pdbs {
 			_, err := context.clientSet.PolicyV1beta1().PodDisruptionBudgets(context.ns.Name).Create(pdb)
 			if err != nil {
 				t.Fatalf("Failed to create PDB: %v", err)
 			}
 		}
+		// Wait for PDBs to be stable
+		for i, pdb := range test.pdbs {
+			waitPDBStable(t, context.clientSet, test.pdbPodNum[i], context.ns.Name, pdb.Name)
```
Comment (k82cn) [inline test/integration/scheduler/preemption_test.go]: Should we check pdb stable in scheduler cache directly?

**Comment by k82cn** [comment]
LGTM :)

File: test/integration/scheduler/preemption_test.go (hunk 1 above)
Comment (bsalamat) [reply test/integration/scheduler/preemption_test.go]: Good point. I think that is a safer approach to ensure that PDB is updated when scheduler processes it. I will make the change.

**Review by resouer — COMMENTED** [review]
the main test logic lgtm except for bazel failure :)

**Comment by bsalamat** [comment]
/hold

I am still working on this

**Review by greghaynes — COMMENTED** [review]
I verified this fixes the issue I was encountering in https://github.com/kubernetes/kubernetes/pull/57057. Just a couple of minor comments

File: test/integration/scheduler/preemption_test.go [hunk 2]
```diff
@@ -779,47 +812,77 @@ func TestPDBInPreemption(t *testing.T) {
...
 				},
 			}),
-			preemptedPodIndexes: map[int]struct{}{0: {}, 1: {}},
+			// The third node is chosen because PDB is not violated for node 3 and the victims have lower priority than node-2.
+			preemptedPodIndexes: map[int]struct{}{4: {}, 5: {}, 6: {}},
 		},
 	}
 
 	for _, test := range tests {
+		fmt.Printf("==================BOBBY: running test %v\n", test.description)
```
[the remark this answers is not in the document]
Comment (bsalamat) [reply test/integration/scheduler/preemption_test.go]: Yes. These are part of the "experimenting" commit and that's why I have put a hold on this PR. I will remove them when the PR is ready.
I am seeing scheduler's cache getting out of sync. I have noticed that it happens only when tests are created with a separate "pod informer" (in `initTestSchedulerWithOptions`). I would like to find the root-cause before merging this PR.

[the remark this answers is not in the document]
Comment (greghaynes) [reply test/integration/scheduler/preemption_test.go]: Ah, sorry, I missed the hold :)

**Comment by bsalamat** [comment]
The failed integration test is caused by #63262. This should pass after #63264 is merged.

**Comment by bsalamat** [comment]
I rebased this. It should be good to go now.

**Pull request merged**